DS/DSAssignment7/BinarySearch.java

```diff
@@ -0,0 +1,41 @@
+public class BinarySearch {
+
+	private boolean flag = false;
+	private int position = 0;
+
+	public int binarySearch(int input[], int start, int end, int key) {
+
+		if (start > end) {
+			return (position + 1);
+		}
+
+		else if (start == end) {
+
+			if (input[start] == key) {
+				position = start;
+			} else if (flag == false) {
```

Comment:
Never write conditions this way. If any variable is a boolean then directly use it. e.g. 

if ( ! flag)
{
  .....
}

instead of 
if (flag == false)
{ 
  ...
}